wrappers/python/zxing.cpp

```diff
@@ -106,17 +150,61 @@ Results read_barcodes(py::object _image, const BarcodeFormats& formats, bool try
 	return read_barcodes_impl(_image, formats, try_rotate, try_downscale, text_mode, binarizer, is_pure, ean_add_on_symbol);
 }
 
-Image write_barcode(BarcodeFormat format, std::string text, int width, int height, int quiet_zone, int ec_level)
+class WriteResult
+{
+public:
+	WriteResult(std::vector<char>&& result_data_, std::vector<py::ssize_t>&& shape_)
+	{
+		result_data = result_data_;
+		shape = shape_;
+		ndim = shape.size();
+	}
+
+	WriteResult(WriteResult const &o) : ndim(o.ndim), shape(o.shape), result_data(o.result_data) { }
+
+	const std::vector<char>& get_result_data() const { return result_data; }
+
+	static py::buffer_info get_buffer(const WriteResult& wr)
+	{
+		return py::buffer_info(
+			reinterpret_cast<void*>(const_cast<char*>(wr.result_data.data())),
+			sizeof(char), "B", wr.shape.size(), wr.shape, { wr.shape[0], py::ssize_t(1) }
+		);
+	}
+
+	const std::vector<py::ssize_t> get_shape() const { return shape; }
+
+private:
+	py::ssize_t ndim;
+	std::vector<py::ssize_t> shape;
+	std::vector<char> result_data;
+};
+
+py::object write_barcode(BarcodeFormat format, std::string text, int width, int height, int quiet_zone, int ec_level)
 {
 	auto writer = MultiFormatWriter(format).setEncoding(CharacterSet::UTF8).setMargin(quiet_zone).setEccLevel(ec_level);
 	auto bitmap = writer.encode(text, width, height);
 
-	auto result = Image({bitmap.height(), bitmap.width()});
-	auto r = result.mutable_unchecked<2>();
-	for (py::ssize_t y = 0; y < r.shape(0); y++)
-		for (py::ssize_t x = 0; x < r.shape(1); x++)
-			r(y, x) = bitmap.get(narrow_cast<int>(x), narrow_cast<int>(y)) ? 0 : 255;
-	return result;
+	std::vector<char> result_data(bitmap.width() * bitmap.height());
+	for (py::ssize_t y = 0; y < bitmap.height(); y++)
+		for (py::ssize_t x = 0; x < bitmap.width(); x++)
+			result_data[x + (y * bitmap.width())] = bitmap.get(narrow_cast<int>(x), narrow_cast<int>(y)) ? 0 : 255;
+
+
+	WriteResult res(std::move(result_data), {bitmap.height(), bitmap.width()});
+
+	auto resobj = py::cast(res);
+
+	py::dict array_interface;
+	array_interface["version"] = py::cast(3);
+	// TODO: This should be a tuple and is optional
```

Comment:
Can you elaborate on this a bit?

Reply:
As I mentioned earlier, this is actually wrong. PIL already returns a buffer here and that is also part of the `__array_interface__` spec.